draft-davidben-tls-merkle-tree-certs.md

```diff
@@ -434,9 +434,12 @@ A Merkle Tree certification authority is defined by the following values:
 `batch_duration`:
 : A number of seconds which determines how frequently the CA issues certificates. See details below.
 
-`window_size`:
+`validity_window_size`:
 : An integer describing the maximum number of unexpired batches which may exist at a time. This value is determined from `lifetime` and `batch_duration` by `floor(lifetime / batch_duration) + 1`.
 
+`storage_window_size`:
+: An integer describing the number of latest issued batches the CA is guaranteed to serve. `validity_window_size` MUST be at least twice `validity_window_size`.
```

Comment:
Does a storage window apply to the mirrors too? If not, I don't think it's possible to set up a new mirror for a CA that already exists.

To that end, I think we'd need to update the procedure for mirrors to account for this, as the current procedure assumes the CA is obligated to go back to batch number zero.

Reply:
You mean apply to TS? For now we could say a TS is allowed to return 410 as it desires? (And have browsers or another document stipulate stricter requirements for the different use cases of TS?)

Reply:
Yeah, I mean that the procedure for the TS, given non-infinite storage windows, is actually kinda tricky. A new TS cannot even run the algorithm in {{single-trusted-service}} because `old_latest_batch` will be zero, and the CA won't serve the old ones.

If a TS is allowed to return 410 whenever it feels like it, that kinda defeats the purpose of a TS I think. The TS is the one that's trusted to _not_ randomly stop serving things. :-)

Reply:
Giving them the same storage window seems reasonable. Though whatever we choose will need to be reflected into the update algorithm I mentioned.

Reply:
Done.